sycl/source/detail/platform_impl.cpp

```diff
@@ -140,8 +166,17 @@ static void filterWhiteList(vector_class<RT::PiDevice> &pi_devices) {
                                       info::device::driver_version>::get(dev);
 
     for (const DevDescT &desc : whiteList) {
-      // At least device name is required field to consider the filter so far
-      if (nullptr == desc.devName ||
+      if (nullptr != desc.pltName &&
+          !std::regex_match(
+              pltName, std::regex(std::string(desc.pltName, desc.pltNameSize))))
```

Comment:
Not for this patch: you can replace std::regex(std::string) constructor with  std::regex(const char*, size_t) constructor.

Reply:
Of course, already have it in the "refactor patch" I'm preparing.